### ISSUE TITLE
Feature: API check sync status of folder[s]

### ISSUE THREAD

**Issue by supersexy** [syncthing/syncthing#6075]
Sorry, I can not find a reliable, supported way to do an api call that reports about the sync status of one or all folder(s). I would like to do something on a folder with a bash script, but only if that folder is synced.

I found a few hints in the forum:
https://docs.syncthing.net/rest/db-completion-get.html looks like a simple `completion < 100` will do it, however my local computer gives me this result for a folder:
```json
{
  "completion": 0,
  "globalBytes": 2027069255,
  "needBytes": 2027069255,
  "needDeletes": 0,
  "needItems": 251738
}
```
Value of `completion` is 0 - why that? So that easy one will not work. Unfortunately I can not find any hint in the docs that helps me.

Then I found https://docs.syncthing.net/rest/db-status-get.html - here I _guess_ that `needBytes` could be a good indicator if a directory is synced, but I am not sure - I would like to not guess.

I have to dig much too deep for such a fundamentally important info - the syncthing api should make it much easier to get the sync status.

So I would like to propose an API call `GET rest/db/sync` that will give simple binary yes/no results about the sync status of all folders if called without options or of one folder if called with an optional folder name.

Meanwhile it would be very lovely if anybody would like to tell me what is the most reliable way to get the current sync status of a folder, thanks!

**Comment by calmh** [comment]
That `completion` API call is for remote devices. It might work if you pass the magic "local device" device ID, but I'm not sure. It's probably guaranteed not to work if you pass your actual device ID or no device ID at all.

On the other hand, the `status` API call is the right one, working as you guessed. We could have a more straightforward API call though. I'm yet undecided. :)

**Comment by supersexy** [comment]
Thanks for the hint about the device id, was not clear that this is not optional. I understood the docs as "shows completion status of either device or folder". But "local device" does not seem to work:
```
"local device": device ID invalid: incorrect length
```
Yes, it would be great to have that information available without too much deep diving into api docs and pitfall-free for users that just want to be sure that everything is in sync and operations can be started on the synced folders (or not). 

Thank you very much for your attention!

**Comment by supersexy** [comment]
But wait, when using `device` as found in `system/status myID` I still get that same output. Second parameter is a folder-id. Is this a bug? Also: this is the only device in the network right now, other devices offline - does this matter?

**Comment by supersexy** [comment]
with `db/status` I need to first get the folders and then make another request for each folder. 

So yes, one new simplified api call that could get that info with just one request would be very welcome. :)

**Comment by Catfriend1** [comment]
Sorry, normally I don't write +1 but this would be a great bless for the Android app. I took a whole week to draft a full use of the events API to detect completion of local folders, remote devices and overall because I had to study multiple endpoints and how SyncthingController.js interprets them. But: I'm also very satisfied what Syncthing already has (it worked like I expected it). It could just offer an extra rest api endpoint for convenience. E.g. where it lists folders and devices in a nested model and each has a completion (0...100) in its json node.
